### PR TITLE
Fix bug introduced in #647

### PR DESCRIPTION
**Pull request by AustinShalit**
As discussed in the hangouts...

### CODE REVIEW

File: core/src/main/java/edu/wpi/grip/core/serialization/Project.java [hunk 1]
```diff
@@ -127,19 +130,42 @@ public void save(File file) throws IOException {
...
     // Only update the flag the save isn't already dirty
     // We don't need to be redundantly checking if the event dirties the save
-    if (!saveIsDirty && dirtySaveEvent.doesDirtySave()) {
-      saveIsDirty = true;
+    if (!saveIsDirty.get() && dirtySaveEvent.doesDirtySave()) {
+      saveIsDirty.set(true);
+    }
+  }
+
+  public class ObservableBoolean {
```
Comment (SamCarlberg) [inline core/src/main/java/edu/wpi/grip/core/serialization/Project.java]: Does this need to be public? Should also make this static, does look like it needs a reference to the Project instance

Comment (JLLeitschuh) [inline core/src/main/java/edu/wpi/grip/core/serialization/Project.java]: Make this `final` as well please

**Comment by JLLeitschuh** [comment]
LGTM

**Comment by AustinShalit** [comment]
LGTM

**Pull request merged**